src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jvm/OracleJDK.kt

```
@@ -25,8 +26,20 @@ class OracleJDK : VersionedJavaDevelopmentKit {
 
     override fun install(connection: SshConnection) {
         download(connection)
-        connection.execute("tar -xzf $jdkArchive")
-        connection.execute("echo '${use()}' >> ~/.bashrc")
+        val execute = connection.execute("cksum jdk-8u131-linux-x64.tar.gz")
+
+        // Checks 3 times if the downloaded archive of JDK has a proper checksum.
+        // This is done in order to prevent failure to execute unzipping a corrupted file.
+        for (attempt in 0..2) {
+            if (execute.output.trim() == CHECKSUM_FROM_ORACLE_JDK_TAR_GZ_OUTPUT) {
```

Also if it fails for the third time I would love to see some logging and an `Exception` thrown. Right now we are ignoring the problem. In this case it is definitely better to fail fast/early with a clear cause provided.